pyannote/audio/utils/powerset.py

```diff
@@ -138,3 +138,59 @@ def to_powerset(self, multilabel: torch.Tensor) -> torch.Tensor:
             torch.argmax(torch.matmul(multilabel, self.mapping.T), dim=-1),
             num_classes=self.num_powerset_classes,
         )
+
+    def permutation_powerset(self, permutation_ml: torch.Tensor) -> torch.Tensor:
```

Comment:
Shouldn't we cache its output to avoid computing this over and over again?
```python
from functools import lru_cache

@lru_cache
def permutation_powerset(...)
```

Reply:
It seems to *slightly* degrade performance in batch mode.
Using `@lru_cache` on what's currently `find_ps_perm`, as i don't think caching `permutation_powerset` make sense since we would be caching batches.

Reply:
Yeah, does not make sense on batches indeed.